src/Microsoft.OData.Client/Serialization/ODataPropertyConverter.cs

```diff
@@ -66,9 +66,58 @@ internal IEnumerable<ODataProperty> PopulateProperties(object resource, string s
                 }
             }
 
+            // Process non-structured dynamic properties stored in the open type dictionary

```

Comment:
What happens to the structured type dynamic properties (i.e complex types) ?

Reply:
@paulodero In the `WriteEntry` method of `Serializer` class [here](https://github.com/OData/odata.net/blob/a398670872ef5694cafe0f5630537b47b911b66e/src/Microsoft.OData.Client/Serialization/Serializer.cs#L309), you'll notice a call to `PopulateProperties` and another call to `WriteNestedComplexProperties`. `PopulateProperties` only deals with non-structured types while `WriteNestedComplexProperties` deals with structured types (specifically complex types). Dynamic complex type properties are handled in a similar way as declared complex type properties. The call is made [here](https://github.com/OData/odata.net/pull/1712/files#diff-555a2001db6de39abd06516406c5fdc3R212)

Reply:
I get it John. Thanks.